### PR TITLE
add h3 misc functions - part 2

### PR DESCRIPTION
**Pull request by bharatnc**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
 
Add new h3 miscellaneous functions: `edgeLengthKm`,`exactEdgeLengthKm`,`exactEdgeLengthM`,`exactEdgeLengthRads`,`numHexagons`.

Detailed description / Documentation draft:

Add new h3 miscellaneous functions: `edgeLengthKm`,`exactEdgeLengthKm`,`exactEdgeLengthM`,`exactEdgeLengthRads`,`numHexagons`.
 as per: https://h3geo.org/docs/api/misc/

Splitting up adding the functions into smaller PRs so that it's easy to review.

Related to: #17708

### CODE REVIEW

**Comment by bharatnc** [comment]
Failing tests don't seem to be related.

**Review by kitaisreal — CHANGES_REQUESTED** [review]
Need to make small fixes. Then ready to merge.

File: src/Functions/h3HexAreaKm2.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ class FunctionH3HexAreaKm2 final : public IFunction
 
         for (size_t row = 0; row < input_rows_count; ++row)
         {
-            const UInt64 resolution = data[row];
+            const int resolution = data[row];
```
[the remark this answers is not in the document]
Comment (bharatnc) [reply src/Functions/h3HexAreaKm2.cpp]: Changed to use `UInt8` everywhere.

File: src/Functions/h3EdgeLengthM.cpp [hunk 2]
```diff
@@ -65,7 +65,7 @@ class FunctionH3EdgeLengthM : public IFunction
 
         for (size_t row = 0; row < input_rows_count; ++row)
         {
-            const UInt64 resolution = col_hindex->getUInt(row);
+            const int resolution = col_hindex->getUInt(row);
```
[the remark this answers is not in the document]
Comment (bharatnc) [reply src/Functions/h3EdgeLengthM.cpp]: somehow missed this doing this here and in one other place. Fixed.

**Review by kitaisreal — APPROVED** [review]

**Comment by bharatnc** [comment]
Fixed [build issues](https://s3.amazonaws.com/clickhouse-builds/33621/6f9b82dd2c98ef70ef053eda2602d33bc85d6eb9/clickhouse_build_check_(actions)/report.html), looks like fmtlib doesn't allow mixing int types while formatting the exception.
```
2022-01-21 17:32:46 In file included from ../src/Functions/h3HexAreaKm2.cpp:5:
2022-01-21 17:32:46 In file included from ../src/Columns/ColumnsNumber.h:4:
2022-01-21 17:32:46 In file included from ../src/Columns/ColumnVector.h:4:
2022-01-21 17:32:46 In file included from ../src/Columns/IColumn.h:5:
2022-01-21 17:32:46 In file included from ../src/Common/Exception.h:12:
2022-01-21 17:32:46 In file included from ../contrib/fmtlib-cmake/../fmtlib/include/fmt/format.h:44:
2022-01-21 17:32:46 /build/contrib/fmtlib/include/fmt/core.h:996:5: error: static_assert failed due to requirement 'std::is_same<char8_t, char>::value || std::is_same<char8_t, char>::value' "mixing character types is disallowed"
2022-01-21 17:32:46     static_assert(
2022-01-21 17:32:46     ^
2022-01-21 17:32:46 /build/contrib/fmtlib/include/fmt/core.h:1273:32: note: in instantiation of function template specialization 'fmt::detail::arg_mapper<fmt::basic_format_context<fmt::detail::buffer_appender<char>, char>>::map<char8_t, 0>' requested here
2022-01-21 17:32:46   return arg_mapper<Context>().map(val);
...                   ^
```

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**